src/vs/workbench/contrib/codeEditor/browser/saveParticipants.ts

```diff
@@ -287,9 +287,26 @@ class CodeActionOnSaveParticipant implements ITextFileSaveParticipant {
 			? setting
 			: Object.keys(setting).filter(x => setting[x]);
 
-		const codeActionsOnSave = settingItems
+		if (!settingItems.length) {
+			return undefined;
+		}
+
+		const excludedActions = Array.isArray(setting)
+			? []
+			: Object.keys(setting)
+				.filter(x => setting[x] === false)
+				.map(x => new CodeActionKind(x));
+
+		let codeActionsOnSave = settingItems
 			.map(x => new CodeActionKind(x));
 
+		if (!excludedActions.some(a => a.equals(CodeActionKind.SourceFixAll))) {
```

Comment:
I don't think we should hardcode `FixAll` here. Instead, the logic should be:

- If there are multiple code actions on save
- Remove any of them that are strict subsets of the other ones (such as `source.fixAll.eslint` being a subset of `source.fixall`)
- Then o

Reply:
@mjbvz Thank you for the comment. I've updated the code to remove subsets from settingItems.